addons/mail/controllers/mail.py

```diff
@@ -197,3 +201,55 @@ def mail_action_unfollow(self, model, res_id, pid, token, **kwargs):
             'model_name': request.env['ir.model'].sudo()._get(model).display_name,
             'access_url': record._notify_get_action_link('view', model=model, res_id=res_id) if display_link else False,
         })
+
+    @http.route("/mail/message/redirect", methods=["POST"], type="json", auth="public")
+    @add_guest_to_context
```

Comment:
The thing is that here we won't have any `message_id` following the route. So, it would be something like `/mail/message` only. Which was contradicting in `mock_server` with other routes like `mail/message/post`.

Now, in a POST method I don't think we should pass the `message_id` in the route itself.

Reply:
Yes, looks good for me to pass `message_id` in route. This has no reason to be a `POST` as the route simply retrieve data.